aerleon/lib/naming.py

```diff
@@ -877,11 +909,26 @@ def _ParseYamlNetworks(self, file_data: Dict[str, Any], file_name: str) -> None:
                     continue
 
                 if comment is None:
-                    unit.items.append(value)
+                    if network_ref:
+                        addr_unit.items.append(value)
```

Comment:
I spent some time looking at how Naming.GetNet will behave when a network refers to a network with FQDN data, or when a network contains mixed data. Mostly the behavior looks reasonable, but this is complex enough that some extra unit tests are needed.

Some cases worth testing:
* A network which references only FQDN-only networks.
* A network which references networks with a mix of FQDN-only, mixed, and IP-only networks.
* A network with only FQDN data.
* A network with mixed data (FQDN and IP items).

I also spotted a corner case that might surprise users: in the following example, GetNet('EXAMPLE_NETWORK') returns an empty list, which the generators will interpret to mean that `source-address` was not present at all in the term. In this example, source-address is not included at all in the generated config. Would it be better to crash in this case? Or generate a warning?

```
terms:
- name: example-term
  source-address: EXAMPLE_NETWORK
...
networks:
  EXAMPLE_NETWORK:
    fqdn: example.com
```

Reply:
Should have all the cases covered for FQDN and other changes.

Reply:
Yeah, and I like that we are going to crash now for GetNet / GetFQDN calls that produce no results.